src/main/java/com/google/codeu/servlets/LoginServlet.java

```diff
@@ -22,30 +22,106 @@
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
+import javax.servlet.ServletException;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import com.google.codeu.data.Datastore;
+import com.google.codeu.data.User;
 
 /**
  * Redirects the user to the Google login page or their page if they're already logged in.
  */
 @WebServlet("/login")
 public class LoginServlet extends HttpServlet {
 
+  private Datastore datastore;
+
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+  public void init() {
+    datastore = new Datastore();
+  }
+
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+
+    HttpSession session = request.getSession();
+
+    //if user tries to login again just redirect them to their pages
+    // if they aren't in the database yet add them to the database
+    if(session.getAttribute("logged_in") != null){
+      String email = (String) session.getAttribute("email");
+      if(!email.equals("")){
+        User user = datastore.getUser(email);
+        if(user == null){
+          datastore.storeUser(user);
+        }
 
-    UserService userService = UserServiceFactory.getUserService();
+        response.sendRedirect("user-page.html?user=" + email);
+        return;
+      }
+    }
+
+    request.setAttribute("error", false);
+
+    String queryParam = request.getQueryString();
+    if(queryParam != null && queryParam.length() > 0){
+      int errorNumber = Integer.parseInt(request.getParameter("err"));
+      request.setAttribute("error", true);
+
+      switch(errorNumber){
+        case SignUpServlet.NULL_EMAIL_ERROR:
+          request.setAttribute("error_message", "Email cannot be null");
+          break;
+        case SignUpServlet.NULL_PASSWORD_ERROR:
+          request.setAttribute("error_message", "Password cannot be null");
+          break;
+        case SignUpServlet.INVALID_PASSWORD_ERROR:
+          request.setAttribute("error_message", "Invalid password try again");
+          break;
+        default:
+          break;
+      }
+    }
+
+    request.getRequestDispatcher("WEB-INF/jsp/login.jsp").forward(request, response);
+  }
 
-    // If the user is already logged in, redirect to their page
-    if (userService.isUserLoggedIn()) {
-      String user = userService.getCurrentUser().getEmail();
-      response.sendRedirect("/user-page.html?user=" + user);
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    String email = request.getParameter("email");
+    String password = request.getParameter("password");
+
+
+    if(email.equals("")){
+      response.sendRedirect("/login" + "?err=1");
+      return;
+    }else if(password.equals("")){
```

Comment:
Nit: Since the if ends with return, this can be another if instead of an else if.